TestScatterRegions is unstable

## Flaky Test
TestScatterRegions

### Which jobs are failing
```
2023-09-19T03:11:12.0251518Z [2023/09/19 03:09:43.999 +00:00] [INFO] [labeler.go:69] ["region labeler GC stopped"]
2023-09-19T03:11:12.0252042Z [2023/09/19 03:09:49.549 +00:00] [INFO] [labeler.go:69] ["region labeler GC stopped"]
2023-09-19T03:11:12.0252689Z --- FAIL: TestScatterRegions (6.88s)
2023-09-19T03:11:12.0253161Z     region_scatterer_test.go:147: 
2023-09-19T03:11:12.0254220Z         	Error Trace:	/home/runner/work/pd/pd/pkg/schedule/scatter/region_scatterer_test.go:147
2023-09-19T03:11:12.0255317Z         	            				/home/runner/work/pd/pd/pkg/schedule/scatter/region_scatterer_test.go:69
2023-09-19T03:11:12.0255858Z         	Error:      	"7" is not greater than or equal to "7.5"
2023-09-19T03:11:12.0256276Z         	Test:       	TestScatterRegions
2023-09-19T03:11:12.0256659Z [2023/09/19 03:09:50.302 +00:00] [INFO] [labeler.go:69] ["region labeler GC stopped"]
```

### CI link
https://github.com/tikv/pd/actions/runs/6230663747/job/16911029516?pr=7092

### Reason for failure (if possible)


### Anything else

--- a/pkg/schedule/scatter/region_scatterer.go
+++ b/pkg/schedule/scatter/region_scatterer.go
@@ -451,7 +451,7 @@ func (r *RegionScatterer) selectNewPeer(context engineContext, group string, pee
 	originStorePickedCount := uint64(math.MaxUint64)
 	for _, store := range stores {
 		storeCount := context.selectedPeer.Get(store.GetID(), group)
-		if store.GetID() == peer.GetId() {
+		if store.GetID() == peer.GetStoreId() {
 			originStorePickedCount = storeCount
 		}
 		// If storeCount is equal to the maxStoreTotalCount, we should skip this store as candidate.

--- a/pkg/schedule/scatter/region_scatterer_test.go
+++ b/pkg/schedule/scatter/region_scatterer_test.go
@@ -67,6 +67,7 @@ func TestScatterRegions(t *testing.T) {
 	scatter(re, 5, 50, true)
 	scatter(re, 5, 500, true)
 	scatter(re, 6, 50, true)
+	scatter(re, 7, 71, true)
 	scatter(re, 5, 50, false)
 	scatterSpecial(re, 3, 6, 50)
 	scatterSpecial(re, 5, 5, 50)
@@ -132,20 +133,36 @@ func scatter(re *require.Assertions, numStores, numRegions uint64, useRules bool
 			}
 		}
 	}
+	maxStorePeerTotalCount := uint64(0)
+	minStorePeerTotalCount := uint64(math.MaxUint64)
 
 	// Each store should have the same number of peers.
 	for _, count := range countPeers {
-		re.LessOrEqual(float64(count), 1.1*float64(numRegions*3)/float64(numStores))
-		re.GreaterOrEqual(float64(count), 0.9*float64(numRegions*3)/float64(numStores))
+		if count > maxStorePeerTotalCount {
+			maxStorePeerTotalCount = count
+		}
+		if count < minStorePeerTotalCount {
+			minStorePeerTotalCount = count
+		}
 	}
+	re.LessOrEqual(maxStorePeerTotalCount-minStorePeerTotalCount, uint64(1))
 
 	// Each store should have the same number of leaders.
 	re.Len(countPeers, int(numStores))
 	re.Len(countLeader, int(numStores))
+
+	maxStoreLeaderTotalCount := uint64(0)
+	minStoreLeaderTotalCount := uint64(math.MaxUint64)
 	for _, count := range countLeader {
-		re.LessOrEqual(float64(count), 1.1*float64(numRegions)/float64(numStores))
-		re.GreaterOrEqual(float64(count), 0.9*float64(numRegions)/float64(numStores))
+		if count > maxStoreLeaderTotalCount {
+			maxStoreLeaderTotalCount = count
+		}
+		if count < minStoreLeaderTotalCount {
+			minStoreLeaderTotalCount = count
+		}
 	}
+	// Since the scatter leader depends on the scatter result of the peer, the maximum difference is 2.
+	re.LessOrEqual(maxStoreLeaderTotalCount-minStoreLeaderTotalCount, uint64(2))
 	re.GreaterOrEqual(noNeedMoveNum, 0)
 }
 

